stackslib/src/net/download/nakamoto/tenure_downloader_set.rs

```diff
@@ -472,15 +553,23 @@ impl NakamotoTenureDownloaderSet {
                     &naddr, &downloader.tenure_id_consensus_hash
                 );
                 finished.push(naddr.clone());
-                finished_tenures.push(downloader.tenure_id_consensus_hash.clone());
+                finished_tenures.push(CompletedTenure::from(downloader));
                 continue;
             }
             debug!(
                 "Send request to {} for tenure {} (state {})",
                 &naddr, &downloader.tenure_id_consensus_hash, &downloader.state
             );
             let Ok(sent) = downloader.send_next_download_request(network, neighbor_rpc) else {
-                debug!("Downloader for {} failed; this peer is dead", &naddr);
+                info!(
+                    "Downloader for tenure {} to {} failed; this peer is dead",
+                    &downloader.tenure_id_consensus_hash, &naddr
+                );
+                Self::mark_failure(
```

Comment:
Is there ever a time that we don't call "mark_failure" at the same time as "mark_deprioritized"? Maybe they should be combined to ensure we don't miss one without the other?

Reply:
Sure, I can do that.